### PR TITLE
adds monkey cube suicide

### PR DESCRIPTION
**Pull request by tmtmtl30**
## About The Pull Request

if you try to commit suicide with a monkey cube, it gibs you and spawns a monkey (also works with gorilla cubes). see, monkey cubes activate in water, and when you think about it, stomach acid is really just spicy water. so if you swallow a monkey cube...

also: the suicide fails for species with TRAIT_NOHUNGER -- plasmamen, ethereals, skeletons, abductors, etc., since their biology doesn't involve stomach acid / saliva and thus they can't activate the monkey cube

## Why It's Good For The Game

suicides are nice flavor, and it felt weird that monkey cubes didn't have a fitting suicide

## Changelog
:cl:
add: Distraught employees on Nanotransen stations have been observed consuming monkey cubes as a means of suicide. Nanotransen reminds employees to remain upbeat, and to not use monkey cubes for anything but their intended purpose.
/:cl:

### CODE REVIEW

**Comment by kriskog** [comment]
Why can you eat monkey cubes safely then?

**Comment by RaveRadbury** [comment]
> Why can you eat monkey cubes safely then?

Because you bite them and that breaks up the whole deal. Swallowing one whole keeps it intact.

**Comment by kriskog** [comment]
Seems like it would be like eating lots of uncooked pasta

File: code/modules/food_and_drinks/food/snacks_meat.dm [hunk 1]
```diff
@@ -179,6 +179,21 @@
...
+	sleep(10)
+	playsound(M.loc,'sound/items/eatfood.ogg', rand(10,50), TRUE)
+	if(HAS_TRAIT(M, TRAIT_NOHUNGER))
+		to_chat(user, "<span class='warning'>Your body won't activate [src]...</span>")
+		return SHAME
+	sleep(10)
+	Expand()
+	M.visible_message("<span class='danger'>[M]'s torso bursts open as a primate emerges!</span>")
+	ADD_TRAIT(M, TRAIT_DISFIGURED, TRAIT_GENERIC)
+	M.gib(null, TRUE, null, TRUE)
```
Comment (bobbahbrown) [inline code/modules/food_and_drinks/food/snacks_meat.dm]: why add disfigured if we just gib them

Comment (tmtmtl30) [reply code/modules/food_and_drinks/food/snacks_meat.dm]: most of my code is cobbled together from other, similar places, to prevent me missing anything, so i haven't the faintest idea

File: code/modules/food_and_drinks/food/snacks_meat.dm [hunk 2]
```diff
@@ -179,6 +179,20 @@
 		visible_message("<span class='notice'>[src] fails to expand!</span>")
 	qdel(src)
 
+/obj/item/reagent_containers/food/snacks/monkeycube/suicide_act(mob/user)
+	var/mob/living/M = user
+	user.visible_message("<span class='suicide'>[M] is putting [src] in [M.p_their()] mouth! It looks like [M.p_theyre()] trying to commit suicide!</span>")
+	sleep(10)
```
Comment (nemvar) [inline code/modules/food_and_drinks/food/snacks_meat.dm]: use a do_after here.

File: code/modules/food_and_drinks/food/snacks_meat.dm [hunk 3]
```diff
@@ -179,6 +179,20 @@
...
+	playsound(M.loc,'sound/items/eatfood.ogg', rand(10,50), TRUE)
+	if(HAS_TRAIT(M, TRAIT_NOHUNGER))
+		to_chat(user, "<span class='warning'>Your body won't activate [src]...</span>")
+		return SHAME
+	sleep(10)
+	Expand()
+	M.visible_message("<span class='danger'>[M]'s torso bursts open as a primate emerges!</span>")
+	M.gib(null, TRUE, null, TRUE)
+	return MANUAL_SUICIDE
```
Comment (nemvar) [inline code/modules/food_and_drinks/food/snacks_meat.dm]: This is shit. It doesn't even remove the cube from their hands. You could just throw it away and it will emerge in another room while you still get gibbed. Shit flavour implementation.
Also needs QDELETED checks for src and M could runtime otherwise.

File: code/modules/food_and_drinks/food/snacks_meat.dm (hunk 2 above)
Comment (tmtmtl30) [reply code/modules/food_and_drinks/food/snacks_meat.dm]: aight

File: code/modules/food_and_drinks/food/snacks_meat.dm (hunk 3 above)
Comment (tmtmtl30) [reply code/modules/food_and_drinks/food/snacks_meat.dm]: other delayed suicide acts didn't have checks for the item still being in the user's hands, so i didn't consider (re)moving it the item. not sure where to put the qdeleted check for src, as Expand() qdeletes it anyway (and, this isn't an antagonistic question, but one so i can better understand when and where i need to put qdeleted checks: why do i need to check if M has been qdeleted? it doesn't happen in every proc after every call to an atom, so when is it relevant? just after waits/long computations? what causes the runtime?).

Comment (nemvar) [reply code/modules/food_and_drinks/food/snacks_meat.dm]: qdeleted checks go directly after the sleep. The cube can't expand if it has been deleted while this proc has been sleeping. Same goes the mob.
If they have been deleted while this was on ice, the value in those variables will be null and the proc calls will runtime.

**Comment by ExcessiveUseOfCobblestone** [comment]
I can almost afford the suicide monky

**Comment by Shadowflame909** [comment]
Monkey Vore got removed

Now the monkeys are getting vengance

**Comment by tmtmtl30** [comment]
hopefully should be in line with nemvar's review now

**Comment by Cyberboss** [comment]
>stomach acid is really just spicy water
My sides

File: code/modules/food_and_drinks/food/snacks_meat.dm [hunk 4]
```diff
@@ -179,6 +179,32 @@
 		visible_message("<span class='notice'>[src] fails to expand!</span>")
 	qdel(src)
 
+/obj/item/reagent_containers/food/snacks/monkeycube/suicide_act(mob/user)
+	var/mob/living/M = user
```
Comment (Cyberboss) [inline code/modules/food_and_drinks/food/snacks_meat.dm]:
```suggestion
/obj/item/reagent_containers/food/snacks/monkeycube/suicide_act(mob/living/M)
```

File: code/modules/food_and_drinks/food/snacks_meat.dm [hunk 5]
```diff
@@ -179,6 +179,32 @@
...
+	if(QDELETED(M)) //qdeletion: the nuclear option of self-harm
+		return SHAME
+	if(!eating_success || QDELETED(src)) //checks if src is gone or if they failed to wait for a second
+		M.visible_message("<span class='suicide'>[M] chickens out!</span>")
+		return SHAME
+	if(HAS_TRAIT(M, TRAIT_NOHUNGER)) //plasmamen don't have saliva/stomach acid
+		M.visible_message("<span class='suicide'>[M] realizes [M.p_their()] body won't activate [src]!</span>"
+		,"<span class='warning'>Your body won't activate [src]...</span>")
+		return SHAME
+	playsound(M.loc,'sound/items/eatfood.ogg', rand(10,50), TRUE)
```
Comment (Cyberboss) [inline code/modules/food_and_drinks/food/snacks_meat.dm]:
```suggestion
	playsound(M, 'sound/items/eatfood.ogg', rand(10,50), TRUE)
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by JJRcop** [comment]
It's good practice to try and understand what your code does if you've lifted it from somewhere else. Ultimately you are responsible to know what it's doing as a PR author.

**Comment by tmtmtl30** [comment]
i mean, i understand what all of the code _does_. but there can be situations where something needs to be added, and i don't _know_ that, and then down the line the code starts bugging out. sometimes you need weird boilerplate code to stop something from breaking, so i base the code i create heavily off of other code that already exists in the codebase and that i'm fairly certain works. it's already in, it's already been extensively livetested, so presumably it's in working enough order that if _i_ use it, or pieces of it, it won't break on me. not understanding _why_ something is done != not understanding _what_ something _does_.

**Comment by tmtmtl30** [comment]
hm, it says cyberboss' review is still pending even though i answered it. strange

File: code/modules/food_and_drinks/food/snacks_meat.dm [hunk 6]
```diff
@@ -179,6 +179,31 @@
...
+	if(!eating_success || QDELETED(src)) //checks if src is gone or if they failed to wait for a second
+		M.visible_message("<span class='suicide'>[M] chickens out!</span>")
+		return SHAME
+	if(HAS_TRAIT(M, TRAIT_NOHUNGER)) //plasmamen don't have saliva/stomach acid
+		M.visible_message("<span class='suicide'>[M] realizes [M.p_their()] body won't activate [src]!</span>"
+		,"<span class='warning'>Your body won't activate [src]...</span>")
+		return SHAME
+	playsound(M, 'sound/items/eatfood.ogg', rand(10,50), TRUE)
+	M.temporarilyRemoveItemFromInventory(src) //removes from hands, keeps in M
+	sleep(15) //you've eaten it, you can run now
```
Comment (spookydonut) [inline code/modules/food_and_drinks/food/snacks_meat.dm]: use a timer+callback

Comment (tmtmtl30) [reply code/modules/food_and_drinks/food/snacks_meat.dm]: is there a specific reason why i shouldn't use sleep()? just curious, i'll do it either way

Comment (Cyberboss) [reply code/modules/food_and_drinks/food/snacks_meat.dm]: Someone should write an essay why that I can store in my GitHub auto replies.

TL;DR there's a global queue for sleeping and BYOND doesn't handle it well. Our timer system works around that for much better performance

**Comment by tmtmtl30** [comment]
it should be good now

**Comment by tmtmtl30** [comment]
goddammit github-actions bot i already ADDRESSED the reviews github is just throwing a fit
